src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -701,5 +684,19 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
             _pool.Free(arms);
             return result;
         }
+
+        private PatternSyntax ParseListPattern(bool whenIsKeyword)
+        {
+            ParseSubpatternList(
+                out SyntaxToken openBracket,
+                out SeparatedSyntaxList<PatternSyntax> list,
+                out SyntaxToken closeBracket,
+                SyntaxKind.OpenBracketToken,
+                SyntaxKind.CloseBracketToken,
+                static p => p.ParsePattern(Precedence.Conditional));
+            TryParseSimpleDesignation(out VariableDesignationSyntax designation, whenIsKeyword);
+            var result = _syntaxFactory.ListPattern(openBracket, list, closeBracket, designation);
+            return CheckFeatureAvailability(result, MessageID.IDS_FeatureListPattern);
```

Comment:
Not related to this PR, nit: we may want to do the same for slice pattern. 
Currently the LangVer diagnostic is only reported on `..` portion of slice pattern, but it would be consistent to report it on the whole node. 
Consider leaving a PROTOTYPE note.

Reply:
Agreed. However, that is going to overlap with the list anyways. Should we even produce a separate diagnostic for `..`?

Reply:
Good point. It'd be okay to remove the LangVer diagnostic on `..`.